### PR TITLE
Bump laminas-diactoros, psr/http-message versions

### PR DESCRIPTION
**Pull request by erikn69**
Updates the requirements on [laminas/laminas-diactoros](https://github.com/laminas/laminas-diactoros) to permit the latest version.

>3.0.0
Release Notes for [3.0.0](https://github.com/laminas/laminas-diactoros/milestone/4)
Added
This version adds support for PSR-7 v2.0. While we had previously added return type hints in Diactoros, there were a few cases where PSR-7 updated the interfaces to use void returns where we had not. As a result, this release represents a BC break for anybody extending our classes, as defining an extension that uses one of these methods with a mixed return type (or omitting the return type entirely) will result in a type error.
>
>The interfaces affected were Psr\Http\Message\StreamInterface and Psr\Http\Message\UploadedFileInterface.
>
>For consumers, this version should be entirely backwards compatible in usage. For those extending our classes, you will only run into issues when extending implementations of the above interfaces.

### CODE REVIEW

**Review by parallels999 — APPROVED** [review]

**Comment by parallels999** [comment]
@Sephster ping

File: composer.json [hunk 1]
```diff
@@ -9,14 +9,14 @@
         "league/event": "^2.2",
         "league/uri": "^6.7",
         "lcobucci/jwt": "^4.3 || ^5.0",
-        "psr/http-message": "^1.0.1",
+        "psr/http-message": "^1.0.1 || ^2.0",
```
Comment (Sephster) [inline composer.json:12]: Could we just change this to use 2.0? Looks like it will support all our supported versions of PHP

File: composer.json [hunk 2]
```diff
@@ -9,14 +9,14 @@
...
         "defuse/php-encryption": "^2.3",
         "ext-json": "*",
         "lcobucci/clock": "^2.2 || ^3.0"
     },
     "require-dev": {
         "phpunit/phpunit": "^9.6.6",
-        "laminas/laminas-diactoros": "^2.24.0",
+        "laminas/laminas-diactoros": "^2.24.0 || ^3.0.0",
```
Comment (Sephster) [inline composer.json:19]: Could we just change this to use 3.0.0?

**Review by Sephster — CHANGES_REQUESTED** [review]

**Comment by Sephster** [comment]
Thanks for the PR @erikn69  - just some minor comments from me

File: composer.json (hunk 2 above)
Comment (erikn69) [reply composer.json:19]: Yes, without problem
https://github.com/laminas/laminas-diactoros/blob/2515f4134258b1b418c23cb86606b8a09dd01aea/composer.json#L39

File: composer.json (hunk 1 above)
Comment (erikn69) [reply composer.json:12]: very soon, an update is still missing for it to be available, here
https://github.com/thephpleague/uri-src/issues/108#issuecomment-1510437939

File: composer.json (hunk 2 above)
Comment (Sephster) [reply composer.json:19]: Awesome. Would you be able to make this change please @erikn69 and then I will merge in. If you could add a changelog entry too that would be great but if you don't, no worries. I can add in once merged. Thank you!

Comment (Sephster) [reply composer.json]: I've updated this now. Thanks

**Comment by driesvints** [comment]
@Sephster can we maybe release this as a patch release? We're getting so much issues on Passport about this 😅

**Comment by driesvints** [comment]
Thanks!

**Pull request merged**